### ISSUE TITLE
install package for  for asynchronous processing?

### ISSUE THREAD

**Issue by DanWismer** [NCC-CNC/wheretowork#377]
@jeffreyhanson, do you recall why this is needed?

https://github.com/NCC-CNC/wheretowork/blob/316b2567ecc5a7d35572994490c34ec40d2403df/dev/run_dev.R#L11-L14

**Comment by jeffreyhanson** [comment]
If I recall correctly, when using asynchronous processing via the future package, it doesn't run R using same renv library/packages by default, and so something was needed to ensure that it would use the renv library/packages. I might have that wrong though?

**Comment by DanWismer** [comment]
Thanks Jeff. 

`?renv::install` says it doesn't respect the lockfile, instead it installs the latest versions available from CRAN.  I did update to renv 1.0.7, so I am wondering if that is a new behavior.

**Comment by jeffreyhanson** [comment]
Nah, I think `renv::install()` always did that. I think the reason I used `renv::install()` was because I had issues getting the the asynch worker to run with the renv version of R packages, so as a fallback, I was able to get it to at least work with the CRAN version of the packages.

**Comment by DanWismer** [comment]
@jeffreyhanson do you know why it requires wheretowork package to be installed?

![image](https://github.com/user-attachments/assets/fb0dba76-35cd-4ddf-83d2-ac09759744b6)

**Comment by jeffreyhanson** [comment]
Yeah, we install the packages in renv into the default R package library because the async workers don't use the renv package library and instead use the default R package library. This is definitely not ideal, but I couldn't find another way to do this at the time. There has been some progress in improving async processing in R since I worked on this though. The mirai package (https://github.com/shikokuchuo/mirai) might have a solution for spawning async workers that load packages from a renv library.

**Comment by DanWismer** [comment]
Thanks @jeffreyhanson .

What seems to be happening is that every time a new package is updated on CRAN (data.table for example), it installs the newest version of the package. This would not happen before and rather just spin the app up. Any thoughts on this. I only noticed it when I updated all those package versions.

![image](https://github.com/user-attachments/assets/57362403-be3e-42ef-8e90-f91aeac71994)
